src/domain/interfaces/services/IDiscordService.ts

```diff
@@ -1,13 +1,17 @@
+import { DiscordGuildScheduledEvent } from "@type/DiscordEventTypes";
+
 export interface IDiscordService<
   M = unknown,
   U = unknown,
   P = unknown,
-  T = unknown
+  T = unknown,
+  E = DiscordGuildScheduledEvent,
```

Comment:
acho que o tipo genérico padrão deve ser unknown seguindo o padrão dos outros

```suggestion
  E = unknown,
```